### PR TITLE
Add e2e-periodic workflow

### PR DESCRIPTION
**Pull request by jluhrsen**
This will run 3 e2e jobs twice a day on master. It will run
with HA and local gateway across each of ipv4, ipv6 and dual
stack.

With these jobs we can better track the health of CI instead
of only looking at jobs that run on PRs which may or may not
be healthy themselves.

Signed-off-by: Jamo Luhrsen <jluhrsen@gmail.com>

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -201,96 +203,46 @@ jobs:
...
-    - name: Load docker image
-      run: |
-        docker load --input image.tar
-
-    - name: kind setup
-      run: |
-        export OVN_IMAGE="ovn-daemonset-f:dev"
-        make -C test install-kind
+      - name: Run e2e tests
+        uses: jluhrsen/ovn-kubernetes/.github/actions/common@master
```
Comment (aojea) [inline .github/workflows/test.yml:207]: Do actions receive environment variables?

Comment (jluhrsen) [reply .github/workflows/test.yml:207]: oh... ignore this commit. I didn't realize I was pushing it to the same PR. I'm still trying to figure this part out in my fork.

Comment (jluhrsen) [reply .github/workflows/test.yml]: ok, PR is updated. please review now. Once I have the de-dup stuff working in my fork, I'll push another PR for that.

**Comment by jluhrsen** [comment]
@trozet @dcbw can we get this in?

File: .github/actions/deploy-cluster-run-tests/action.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: "Deploy Kind cluster and run tests"
+description: "Restores k8s from the cache previously built, installs kind, builds and runs e2e tests and exports logs and reports"
```
Comment (aojea) [inline .github/actions/deploy-cluster-run-tests/action.yml:2]: this does not restore from the cache ... it downloads an artifact that we uploaded previously

File: .github/actions/deploy-cluster-run-tests/action.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+runs:
+  using: "composite"
+  steps:
+  - name: Load docker image
+    run: |
+      docker load --input image.tar
```
Comment (aojea) [inline .github/actions/deploy-cluster-run-tests/action.yml:8]: this step depends of a previous step, that creates the `image.tar` having it separate can break easily

**Comment by jluhrsen** [comment]
ugh.... I honestly don't know why this commit keeps showing up on this PR. I'm trying to work on it in my own repo. Every time I put it on my master branch to test it, it shows up on this PR back over here in ovn-org/ovn-kubernetes.

what am I doing wrong?

**Comment by aojea** [comment]
> ugh.... I honestly don't know why this commit keeps showing up on this PR. I'm trying to work on it in my own repo. Every time I put it on my master branch to test it, it shows up on this PR back over here in ovn-org/ovn-kubernetes.
> 
> what am I doing wrong?

what is your workflow and how do you have configured your remotes?
`git remote -v`

**Comment by jluhrsen** [comment]
> > ugh.... I honestly don't know why this commit keeps showing up on this PR. I'm trying to work on it in my own repo. Every time I put it on my master branch to test it, it shows up on this PR back over here in ovn-org/ovn-kubernetes.
> > what am I doing wrong?
> 
> what is your workflow and how do you have configured your remotes?
> `git remote -v`

normally I like to do small bite size, single commit changes in a local topic branch and test then eventually push as a PR to the upstream fork.

In this case, I need my github actions related changes to eventually end up in my local master branch so that I can test it properly since some github actions functionality only operates on the default branch.

I guess the problem is I made my original PR for this from my own repos master branch. I should have done it from a topic
branch.

here is remote -v:

❯ git remote -v
origin  https://github.com/jluhrsen/ovn-kubernetes.git (fetch)
origin  https://github.com/jluhrsen/ovn-kubernetes.git (push)
upstream        https://github.com/ovn-org/ovn-kubernetes.git (fetch)
upstream        https://github.com/ovn-org/ovn-kubernetes.git (push)

**Comment by aojea** [comment]
> I guess the problem is I made my original PR for this from my own repos master branch. I should have done it from a topic
> branch.

hehe,you are working in master `jluhrsen:master` , ... what I 'm doing  is to work in a branch on my fork and PR from the UI my (fork) master branch

**Comment by jluhrsen** [comment]
> > I guess the problem is I made my original PR for this from my own repos master branch. I should have done it from a topic
> > branch.
> 
> hehe,you are working in master `jluhrsen:master` , ... what I 'm doing is to work in a branch on my fork and PR from the UI my (fork) master branch

ok then, I'll have to be more careful and not push a PR from my master branch.

anyway, can we merge this now?

**Comment by aojea** [comment]
it is fine with me, please @trozet or @dcbw

**Review by trozet — APPROVED** [review]

**Pull request merged**